### DISCUSSION TITLE
Is it possible to build a bootstrap ZIP with a custom package list?

### DISCUSSION THREAD

**Discussion by contourintegrals** [termux/termux-packages#11770] [Q&A] [answered]
So basically, I want to make a bootstrap ZIP where the only things are `coreutils` (so mv,cp,rm...), `bash`, `dash`, `unzip`, `nodejs-lts` and `openjdk-17`. Because of the large size of the last two packages, I simply want to make a bootstrap with ONLY those packages (I don't want apt to be installed – there wouldn't be any `pkg update`, _etc._).

Is this possible? I tried to build the aforementioned packages using `./build-package.sh -a aarch64 <pkgs>` but I don't know how to make a bootstrap out of it. `./scripts/generate-bootstraps.sh` just allows to have **additional** packages along with the list of defaults (`-a`). I'd like to add that the package ID is changed.

Thanks in advance!

**Comment by truboxl** [comment, marked as answer]
https://github.com/termux/termux-packages/blob/94346c60743a4b8272817a802ecf6be17d12f61c/scripts/generate-bootstraps.sh#L429-L474

Delete one of these

**Reply by contourintegrals** [reply]
Okay, so, will this work if I use the script in #11491? That's because I'm building using a custom package name and in #11713 you suggested to use that.

**Reply by truboxl** [reply]
The one in #11491 adds `-c` option so it will not download deb files that still use `com.termux` prefix path and instead use the debs you created

YMMV since I never test changing package ID

I believe the repo package list https://grimler.se/termux/termux-main/dists/stable/main/binary-aarch64/Packages downloaded by `generate-bootsraps.sh` does not contain `com.termux` prefix and only used for resolving package dependencies so my earlier comment about `generate-bootstraps.sh` is dumber is not true

Might need to add an option to create a local copy of this list in the future since it likely will break for new packages that are not yet on the repo

**Reply by contourintegrals** [reply]
@truboxl Okay then, I might as well try and see if removing the calls to `pull_package` actually works :star:
